jazz_events/index.js

```diff
@@ -40,382 +40,449 @@ module.exports.handler = (event, context, cb) => {
-					var event_name_params = {
-						Key: {
-							"EVENT_NAME": {
-								S: event.body.event_name
-							}
-						},
-						TableName: config.event_name_table
-					};
-					dynamodb.getItem(event_name_params, function (err, data) {
-						if (err) {
-							logger.error("error reading event_name from database " + err.message);
-							callback({
-								"code": 500,
-								"message": "error reading event_name from database " + err.message
-							});
-						} else {
-							if (data === undefined || data === null || data.Item === undefined || data.Item === null) {
-								logger.error("Invalid EVENT_NAME. " + event.body.event_name);
-								callback({
-									"code": 400,
-									"message": "Invalid EVENT_NAME. " + event.body.event_name
-								}, null);
-							} else {
-								callback(null, data.Item);
-							}
-						}
-					});
-				},
-				validate_event_handler: function (callback) {
-					var event_handler_params = {
-						Key: {
-							"EVENT_HANDLER": {
-								S: event.body.event_handler
-							}
-						},
-						TableName: config.event_handler_table
-					};
-					dynamodb.getItem(event_handler_params, function (err, data) {
-						if (err) {
-							logger.error("error reading event_handler from database " + err.message);
-							callback({
-								"code": 500,
-								"message": "error reading event_handler from database " + err.message
-							});
-						} else {
-							if (data === undefined || data === null || data.Item === undefined || data.Item === null) {
-								logger.error("Invalid EVENT_HANDLER. " + event.body.event_handler);
-								callback({
-									"code": 400,
-									"message": "Invalid EVENT_HANDLER. " + event.body.event_handler
-								}, null);
-							} else {
-								callback(null, data.Item);
-							}
-						}
-					});
-				},
-				validate_event_status: function (callback) {
-					var event_status_params = {
-						Key: {
-							"EVENT_STATUS": {
-								S: event.body.event_status
-							}
-						},
-						TableName: config.event_status_table
-					};
-					dynamodb.getItem(event_status_params, function (err, data) {
-						if (err) {
-							logger.error("error reading event_status from database " + err.message);
-							callback({
-								"code": 500,
-								"message": "error reading event_status from database " + err.message
-							});
-						} else {
-							if (data === undefined || data === null || data.Item === undefined || data.Item === null) {
-								logger.error("Invalid EVENT_STATUS. " + event.body.event_status);
-								callback({
-									"code": 400,
-									"message": "Invalid EVENT_STATUS. " + event.body.event_status
-								}, null);
-							} else {
-								callback(null, data.Item);
-							}
+function mapGetEventData(result, event){
+	logger.info("Inside mapGetEventData:"+JSON.stringify(result));
+	return new Promise((resolve, reject) => {
+		var events = [];
+		if (result && result.Items) {
+			result.Items.forEach(function (item) {
```

Comment:
Strongly suggest to avoid using forEach unless you have to. Please always try to use map. Perf of map is always better because it does not mutate the array. Map is the recommended way of using loops on arrays/lists.

Reply:
Replaced forEach with map for looping arrays